lib/player.js

```diff
@@ -4624,6 +4625,140 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
+   * application.
+   *
+   * @param {string} uri
+   * @param {string=} mimeType
+   * @return {!Promise.<shaka.extern.Track>}
+   * @export
+   */
+  async addThumbnailsTrack(uri, mimeType) {
+    if (this.loadMode_ != shaka.Player.LoadMode.MEDIA_SOURCE &&
+        this.loadMode_ != shaka.Player.LoadMode.SRC_EQUALS) {
+      shaka.log.error(
+          'Must call load() and wait for it to resolve before adding image ' +
+          'tracks.');
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.RECOVERABLE,
+          shaka.util.Error.Category.PLAYER,
+          shaka.util.Error.Code.CONTENT_NOT_LOADED);
+    }
+
+    if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
+      // This should not happen, but there are browser implementations that may
+      // not support the Track element.
+      shaka.log.error('Cannot add this text when loaded with src=');
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.RECOVERABLE,
+          shaka.util.Error.Category.TEXT,
+          shaka.util.Error.Code.CANNOT_ADD_EXTERNAL_THUMBNAILS_TO_SRC_EQUALS);
+    }
+
+    if (!mimeType) {
+      mimeType = await this.getTextMimetype_(uri);
+    }
+
+    if (mimeType != 'text/vtt') {
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.RECOVERABLE,
+          shaka.util.Error.Category.TEXT,
+          shaka.util.Error.Code.UNSUPPORTED_EXTERNAL_THUMBNAILS_URI,
+          uri);
+    }
+
+    const ContentType = shaka.util.ManifestParserUtils.ContentType;
+
+    const duration = this.manifest_.presentationTimeline.getDuration();
+    if (duration == Infinity) {
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.RECOVERABLE,
+          shaka.util.Error.Category.MANIFEST,
+          shaka.util.Error.Code.CANNOT_ADD_EXTERNAL_THUMBNAILS_TO_LIVE_STREAM);
+    }
+
+    goog.asserts.assert(
+        this.networkingEngine_, 'Need networking engine.');
+    const buffer = await this.getTextData_(uri,
+        this.networkingEngine_,
+        this.config_.streaming.retryParameters);
+
+    const factory = shaka.text.TextEngine.findParser(mimeType);
+    if (!factory) {
+      throw new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.TEXT,
+          shaka.util.Error.Code.MISSING_TEXT_PLUGIN,
+          mimeType);
+    }
+    const TextParser = factory();
+    const time = {
+      periodStart: 0,
+      segmentStart: 0,
+      segmentEnd: duration,
+      vttOffset: 0,
+    };
+    const data = shaka.util.BufferUtils.toUint8(buffer);
+    const cues = TextParser.parseMedia(data, time);
+
+    const references = [];
+    for (const cue of cues) {
+      const imageUri = shaka.util.ManifestParserUtils.resolveUris(
+          [uri], [cue.payload])[0];
+      if (imageUri.includes('#xywh')) {
+        shaka.log.alwaysWarn('Unsupported image uri', imageUri);
+        continue;
+      }
```

Comment:
Doesn't the existing thumbnail system in Shaka Player allow for sprite sheets?  Could you parse these coordinates instead of skipping the reference?

Reply:
Right now the system is set up to support tiles, but not sprites. At the moment we are going to skip it, and as soon as I have a little more time I will add a PR to support it, give me 1-2 months. Are you agree?

Reply:
Okay.  Sounds good.